aptos-move/aptos-gas-meter/src/algebra.rs

```diff
@@ -165,6 +170,16 @@ impl GasAlgebra for StandardGasAlgebra {
         &mut self,
         abstract_amount: impl GasExpression<VMGasParameters, Unit = InternalGasUnit> + Debug,
     ) -> PartialVMResult<()> {
+        if self
```

Comment:
This adds a conditional, a vtable lookup and a call - just mentioning :)

Reply:
should be able to eliminate the conditional (to compile-time) - didn't do it for a draft.
function calls will happen, but calling via the dyn trait I don't know how to easily avoid.

Reply:
Yep - that's why I said "i was just commenting" the full GasMeter design should be dynamically dispached much earlier, with different compile time implementations (like create a meter for specific settings and version, and avoid other dynamic check for a version that happen  literally every time we charge gas :) )